wildlifelicensing/apps/main/admin.py

```diff
@@ -37,6 +37,40 @@ class Meta:
         model = WildlifeLicenceType
         exclude = []
 
+    def clean_application_schema(self):
+        schema = self.cleaned_data['application_schema']
+
+        names = []
+
+        def __check_name(schema):
+            if type(schema) is list:
+                for item in schema:
+                    __check_name(item)
+            elif type(schema) is dict:
+                if 'type' in schema:
+                    if 'name' in schema:
+                        if schema['name'] in names:
+                            raise forms.ValidationError('Duplicate Name %s' % schema.get('name'))
+                        else:
+                            names.append(schema['name'])
+                    else:
+                        raise forms.ValidationError('Missing Name %s with label %s' % (schema.get('type'), schema.get('label')))
+                if 'children' in schema:
+                    for item in schema['children']:
+                        __check_name(item)
```

Comment:
Not important but the `for loop` is not necessary here (done in the `if (type) is list`. `_check_name(schema['children'])` is enough.